File: src/python/day3/__main__.py
import modules.timer as timer
from .puzzle1 import mulall
from .puzzle2 import mulWithInstruction
from .input import input

def run():
    print("puzzle1: mulall:", mulall(input))
    for _ in timer.run(100):
        mulall(input)

    print()

    print("puzzle2: mulWithInstruction:", mulWithInstruction(input))
    for _ in timer.run(100):
        mulWithInstruction(input)

if __name__ == "__main__":
    run()

File: src/go/day3/main.go
package main

import (
	_ "embed"
	"fmt"

	"github.com/dancantos/advent2024/src/go/pkg/timeit"
)

func main() {
	var result1 int
	d1 := timeit.Run(func() {
		result1 = mulall(input)
	})

	var result2 int
	d2 := timeit.Run(func() {
		result2 = mulWithInstruction(input)
	})

	fmt.Printf("Puzzle1 (mulall): %d (in %s)\n", result1, d1)
	fmt.Printf("Puzzle2 (mulWithInstruction): %d (in %s)\n", result2, d2)
}
